oggm/tests/test_models.py

```diff
@@ -4043,6 +4043,86 @@ def test_hydro_ref_area(self, hef_gdir, inversion_params):
         odf['dom_area'] = odf['on_area'] + odf['off_area']
         assert_allclose(odf['dom_area'], odf['dom_area'].iloc[0], rtol=1e-4)
 
+    @pytest.mark.slow
+    def test_hydro_dynamical_spinup(self, hef_gdir, inversion_params):
+
+        gdir = hef_gdir
+        gdir.rgi_date = 1990
+
+        # Add debug vars
+        cfg.PARAMS['store_diagnostic_variables'] = ALL_DIAGS
+        # Needed for this to run
+        cfg.PARAMS['store_model_geometry'] = True
+
+        init_present_time_glacier(gdir)
+        tasks.run_with_hydro(gdir, run_task=tasks.run_dynamic_spinup,
+                             store_monthly_hydro=False,
+                             ref_area_from_y0=True,
+                             output_filesuffix='_spinup')
+        tasks.run_with_hydro(gdir, run_task=tasks.run_from_climate_data,
+                             store_monthly_hydro=False,
+                             init_model_filesuffix='_spinup',
+                             ref_geometry_filesuffix='_spinup',
+                             ref_area_from_y0=True,
+                             output_filesuffix='_run')
+
+        with xr.open_dataset(gdir.get_filepath('model_diagnostics',
+                                               filesuffix='_spinup')) as ds:
+            sel_vars = [v for v in ds.variables if 'month_2d' not in ds[v].dims]
+            odf_hist = ds[sel_vars].to_dataframe()
+            # This is for the mass-conservation check
+            mass_in_snow_t1 = odf_hist['snow_bucket'].iloc[-1]
+            odf_hist = odf_hist.iloc[:-1]
+
+        with xr.open_dataset(gdir.get_filepath('model_diagnostics',
+                                               filesuffix='_run')) as ds:
+            sel_vars = [v for v in ds.variables if 'month_2d' not in ds[v].dims]
+            odf_run = ds[sel_vars].to_dataframe().iloc[:-1]
+
+        assert odf_hist.index[0] == 1971
+        assert odf_hist.index[-1] == 1990
+        assert odf_run.index[0] == 1991
+
+        odf = pd.concat([odf_hist, odf_run])
+
+        # Domain area is constant and equal to the first year
+        odf['dom_area'] = odf['on_area'] + odf['off_area']
+        # assert_allclose(odf['dom_area'], odf['dom_area'].iloc[0], rtol=1e-4)
```

Comment:
@fmaussion I think this line should be uncommented

Reply:
No it should be deleted - the glacier grows a bit during the simulation so this does not work